src/EndpointV2/EndpointV2Middleware.php

```diff
@@ -247,19 +248,32 @@ private function resolveAuthScheme(array $authSchemes) : array
 
         foreach($authSchemes as $authScheme) {
             if (isset(self::$validAuthSchemes[$authScheme['name']])) {
+                switch($authScheme['name']) {
```

Comment:
nit: Could we maybe create a separated method for validating if the authScheme is valid?.  That would make the code more clear. You have a similar implementation in multi-auth PR [here](https://github.com/aws/aws-sdk-php/pull/2891/files#diff-27f526732eeef580d2bed18ed21b00547a25f6cebf14cc91e552b043f72526c3R85). Could you reuse that same method and  in this case just pass Identity as null.

Reply:
I think a separate method is a good idea, but the logic needs to be different for reasons I'll explain elsewhere